templates/maintenance.yml

```diff
@@ -0,0 +1,126 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: image-builder-maintenance
+  annotations:
+    openshift.io/display-name: Image Builder maintenance
+    description: |
+      Cronjob related to maintaining the image-builder database.
+    tags: golang
+    iconClass: icon-shadowman
+    template.openshift.io/provider-display-name: Red Hat, Inc.
+labels:
+  template: image-builder-maintenance
+objects:
+- apiVersion: batch/v1
+  kind: CronJob
+  metadata:
+    labels:
+      service: image-builder
+    name: image-builder-maintenance
+  spec:
+    # run maintenance job at midnight on Tuesdays
+    schedule: 0 0 * * 2
+    concurrencyPolicy: Forbid
+    # don't run if the job doesn't get scheduled within 30 minutes
+    startingDeadlineSeconds: 1800
```

Comment:
As I said this is pretty dangerous - long SQL queries will continue stacking up on RDS, I suggest to pass a context with a deadline that is shorter than this. Also, 30 mins is pretty optimistic, since this is a weekly job I would probably go with several hours e.g. 5 hours deadline, 6 hours for k8s until SIGTERM. Deletions can be extremely slow on tables with constraints, in edge we estimated 1.5 month deletion of 40 million entries from a table and we ended up truncating it on stage.